internal/controller/nodedisruption_controller.go

```diff
@@ -209,40 +214,51 @@ func (ndr *SingleNodeDisruptionReconciler) UpdateStatus(ctx context.Context) err
 	return ndr.Client.Status().Update(ctx, &ndr.NodeDisruption, []client.SubResourceUpdateOption{}...)
 }
 
+func (ndr *SingleNodeDisruptionReconciler) GenerateRejectedStatus(reason string) []nodedisruptionv1alpha1.DisruptedBudgetStatus {
+	return []nodedisruptionv1alpha1.DisruptedBudgetStatus{
+		{
+			Reference: nodedisruptionv1alpha1.NamespacedName{
+				Namespace: ndr.NodeDisruption.Namespace,
+				Name:      ndr.NodeDisruption.Name,
+				Kind:      ndr.NodeDisruption.Kind,
+			},
+			Reason: reason,
+			Ok:     false,
+		},
+	}
+}
+
 // ValidateInternalConstraints check that the Node Disruption is valid against internal constraints
 // such as deadline or constraints on number of impacted nodes
-func (ndr *SingleNodeDisruptionReconciler) ValidateWithInternalConstraints(_ context.Context) (anyFailed bool, statuses []nodedisruptionv1alpha1.DisruptedBudgetStatus) {
+func (ndr *SingleNodeDisruptionReconciler) ValidateWithInternalConstraints(ctx context.Context) (anyFailed bool, statuses []nodedisruptionv1alpha1.DisruptedBudgetStatus, err error) {
 	disruptedNodes := resolver.NewNodeSetFromStringList(ndr.NodeDisruption.Status.DisruptedNodes)
 
 	if ndr.Config.RejectEmptyNodeDisruption && disruptedNodes.Len() == 0 {
-		return true, []nodedisruptionv1alpha1.DisruptedBudgetStatus{
-			{
-				Reference: nodedisruptionv1alpha1.NamespacedName{
-					Namespace: ndr.NodeDisruption.Namespace,
-					Name:      ndr.NodeDisruption.Name,
-					Kind:      ndr.NodeDisruption.Kind,
-				},
-				Reason: "No Node matching selector",
-				Ok:     false,
-			},
+		return true, ndr.GenerateRejectedStatus("No Node matching selector"), nil
+	}
+
+	allDisruptions := &nodedisruptionv1alpha1.NodeDisruptionList{}
+	err = ndr.Client.List(ctx, allDisruptions)
+	if err != nil {
+		return false, nil, err
+	}
+	for _, otherDisruption := range allDisruptions.Items {
+		if otherDisruption.Name == ndr.NodeDisruption.Name {
+			continue
+		}
+		if otherDisruption.Status.State == nodedisruptionv1alpha1.Pending || otherDisruption.Status.State == nodedisruptionv1alpha1.Granted {
+			otherDisruptedNodes := resolver.NewNodeSetFromStringList(otherDisruption.Status.DisruptedNodes)
+			if otherDisruptedNodes.Intersection(disruptedNodes).Len() > 0 {
+				return true, ndr.GenerateRejectedStatus(fmt.Sprintf(`Selected node(s) overlap with another disruption: ”%s"`, otherDisruption.Name)), nil
```

Comment:
This block is executed even if the flag `RejectOverlappingDisruption` is not set, no ? am I missing something?